### PR TITLE
Fixed backward compatibility of checkpoints and arrows marker

### PR DESCRIPTION
**Pull request by FileEX**
Added a new optional argument to createMarker
``ignoreAlphaLimits = false``

This argument is only valid for checkpoint and arrow markers. It defines whether the alpha of these markers can be changed. If the argument is false, the marker has a constant alpha as before, i.e. checkpoint always 128 and arrow always 255. If the argument is true, alpha can be changed.

Default is **false**

I tested this using ``createMarker``, ``setElementAlpha``, ``setMarkerColor`` and ``setMarkerType`` on the server and client side. All my tests were successful

### CODE REVIEW

File: Client/mods/deathmatch/logic/CClient3DMarker.h [hunk 1]
```diff
@@ -45,14 +45,17 @@ class CClient3DMarker : public CClientMarkerCommon
     void SetVisible(bool bVisible) { m_bVisible = bVisible; };
 
     SColor GetColor() const { return m_Color; }
-    void   SetColor(const SColor& color) { m_Color = color; }
+    void   SetColor(const SColor& color);
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/CClient3DMarker.h]: `noexcept`

File: Client/mods/deathmatch/logic/CClientCheckpoint.h [hunk 2]
```diff
@@ -74,6 +74,9 @@ class CClientCheckpoint : public CClientMarkerCommon
     static bool          IconToString(unsigned char ucIcon, SString& strOutString);
     void                 ReCreateWithSameIdentifier();
 
+    void IgnoreAlphaLimits(bool ignore) noexcept { m_ignoreAlphaLimits = ignore; };
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/CClientCheckpoint.h]: What do you think about `SetIgnoreAlphaLimits`?

File: Client/mods/deathmatch/logic/CClientMarkerCommon.h [hunk 3]
```diff
@@ -56,4 +56,7 @@ class CClientMarkerCommon
 
     virtual void StreamIn() = 0;
     virtual void StreamOut() = 0;
+
+    virtual void IgnoreAlphaLimits(bool ignore) = 0;
+    virtual bool AreAlphaLimitsIgnored() const = 0;
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/CClientMarkerCommon.h]: `noexcept`?

**Review by TracerDS — CHANGES_REQUESTED** [review]

**Comment by Dutchman101** [comment]
Please resolve conflicts
@FileEX

**Comment by FileEX** [comment]
> Please resolve conflicts @FileEX

Done

**Pull request merged**